File: src/main/specnaz/src/main/java/org/specnaz/impl/Notifier.java
package org.specnaz.impl;

public interface Notifier {
    void setupStarted();

    void setupSucceeded();

    void setupFailed(Throwable error);

    void teardownStarted();

    void teardownSucceeded();

    void teardownFailed(Throwable error);

    void started(SingleTestCase testCase);

    void passed(SingleTestCase testCase);

    void failed(SingleTestCase testCase, Throwable e);

    void skipped(SingleTestCase testCase, Throwable assumptionViolated);

    void ignored(SingleTestCase test);

    Notifier subgroup(String description);
}

File: src/main/kotlin/org/specnaz/impl/Notifier.kt
package org.specnaz.impl

interface Notifier {
    fun started(test: TestCase)
    fun passed(test: TestCase)
    fun failed(test: TestCase, assertion: AssertionError)
    fun threw(test: TestCase, error: Throwable)
    fun ignored(test: TestCase)
}
